examples/tower_client/src/main.rs

```diff
@@ -0,0 +1,82 @@
+use futures::TryStreamExt;
+use http::header::USER_AGENT;
+use tower::ServiceBuilder;
+use tower_http::{
+    classify::StatusInRangeAsFailures, decompression::DecompressionLayer,
+    set_header::SetRequestHeaderLayer, trace::TraceLayer,
+};
+use twitch_api::{client::TowerService, HelixClient};
+use twitch_oauth2::{AccessToken, AppAccessToken, UserToken};
+
+fn main() {
+    use std::error::Error;
+    if let Err(err) = run() {
+        println!("Error: {err}");
+        let mut e: &'_ dyn Error = err.as_ref();
+        while let Some(cause) = e.source() {
+            println!("Caused by: {cause}");
+            e = cause;
+        }
+    }
+}
+
+#[tokio::main]
+async fn run() -> Result<(), Box<dyn std::error::Error + Send + Sync + 'static>> {
+    let _ = dotenvy::dotenv();
+    tracing_subscriber::fmt()
+        .with_env_filter(tracing_subscriber::EnvFilter::from_default_env())
+        .with_file(true)
+        .with_line_number(true)
+        .init();
+    let mut args = std::env::args().skip(1);
+
+    let tower_client = ServiceBuilder::new()
+        // This buffer is trivial, we need the service to be clonable
+        .buffer(1)
+        // Setup tracing
+        .layer(TraceLayer::new(
+            StatusInRangeAsFailures::new(400..=599).into_make_classifier(),
+        ))
+        .layer(TraceLayer::new_for_http())
+        // Set the user-agent
+        .layer(SetRequestHeaderLayer::overriding(
+            USER_AGENT,
+            twitch_api::client::user_agent(Some("twitch_api/examples/tower_client".parse()?))?,
+        ))
+        .layer(DecompressionLayer::new())
+        // Rate limit to 1 request per second
+        .layer(tower::limit::RateLimitLayer::new(
+            1,
+            std::time::Duration::from_secs(1),
+        ))
```

Comment:
I want to replace this with a rate limiter that knows about twitch rate limits, reading `ratelimit-limit`, `ratelimit-remaining` and `ratelimit-reset`

Reply:
ugh, need a sleep for this to be efficient. Need runtime agnostic async sleep...